controllers/discovery/initDiscoveryUserData.js

```diff
@@ -50,6 +50,8 @@ const InitDiscoveryUserData = async (req, res) => {
     );
     let topicIds = user_topics.map((topic) => topic.topic_id);
 
+    const topicIdQuery = topicIds?.length > 0 ? `AND tp.topic_id in (:topicIds)` : '';
+
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of optional chaining (`?.`) and ternary operator to construct `topicIdQuery` is a good approach to handle the potential null or undefined `topicIds`. This will prevent SQL errors when `topicIds` is null or empty. However, please ensure that the `topicIds` variable is properly sanitized before being used in the SQL query to prevent SQL injection attacks.
```diff
-    const topicIdQuery = topicIds?.length > 0 ? `AND tp.topic_id in (:topicIds)` : '';
+    const topicIdQuery = Array.isArray(topicIds) && topicIds.length > 0 ? `AND tp.topic_id in (:topicIds)` : '';
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->